[8.x] PHP 8.1 Support

driesvints opened the pull request:
This PR contains more PHP 8.1 fixes:

- Remove mimicking PHP 8.0 now that we're nearing the PHP 8.1 release date
- Add `prefer-lowest` builds
- Add separate build step for minimum versions for Symfony, PHPUnit & CommonMark we cannot bump these through Composer
- Bumps some libraries for PHP 8.1 compatibility
- Added more clearer message for skipped mail tests on PHP 8.1 since SwiftMailer won't be patched for it
- Unskipped PHP 8.1 tests

driesvints committed:
Remove platform requirement

driesvints committed:
Enable prefer-lowest builds for PHP 8.1

driesvints committed:
Bump whoops

driesvints committed:
Minimum symfony version

driesvints committed:
Bump carbon

driesvints committed:
Try dev

GrahamCampbell commented on .github/workflows/tests.yml:
```diff
@@ -68,6 +61,14 @@ jobs:
+        with:
+          timeout_minutes: 5
+          max_attempts: 5
+          command: composer require symfony/symfony:5.4.x-dev --no-interaction --no-update
```
why do we need this?

driesvints replied:
PHP 8.1 will only be fully support on Symfony 5.4. This is needed for prefer-lowest builds without bumping the minimum Symfony requirement for non-PHP 8.1 installs.

GrahamCampbell replied:
> PHP 8.1 will only be fully support on Symfony 5.4.

Oh, really? I thought Symfony treated PHP version support as a bug fix?

driesvints replied:
Nicolas confirmed this with me.

driesvints committed:
Require commonmark

driesvints committed:
Add phpunit

driesvints committed:
Bump predis

driesvints committed:
More clearly describe failing mail tests

driesvints committed:
Remove skips for Redis tests

tillkruss commented:
Predis fix merged here: https://github.com/predis/predis/commit/92adf32e88cc7e86e30dd073ecefefff0547fdb3

driesvints committed:
try predis dev

GrahamCampbell commented on composer.json:
```diff
@@ -89,7 +89,7 @@
         "pda/pheanstalk": "^4.0",
         "phpunit/phpunit": "^8.5.19|^9.5.8",
-        "predis/predis": "^1.1.8",
+        "predis/predis": "v1.1.x-dev",
```
`^1.1.9`

driesvints replied:
I'll do that when Till has actually tagged the release.

GrahamCampbell replied:
why? it'll work even without, and it means we don't need to go back and update the code later.

tillkruss replied:
I'd agree with testing `v1.1.x-dev` first before I tag, just in case there is anther minor compat issue.

mallardduck committed:
Make ArgonHasher aware of argon provider (#39046)

* Make ArgonHasher aware of argon provider

If sodium provides the argon2 then threads must be 1, if "standard" (libargon2) is the provider it can be 1 or 2.
Not sure why this seems to only affect PHP 8.1 however as the behaviour seems to date back quite a bit.

https://github.com/php/php-src/blob/2a3760a2d12b1a4c5c0207386b6cae84e404ee7e/ext/sodium/sodium_pwhash.c#L197

* remove skips for Hashing tests

driesvints committed:
Enable tests

driesvints committed:
Re-skip test

driesvints commented:
@GrahamCampbell have patience 😅

driesvints commented:
I will change it once Till has tagged a new release.

driesvints committed:
Adjust skips

driesvints committed:
Re-enable dynamodb tests

driesvints committed:
Bump predis

driesvints committed:
Bump PHP AWS SDK

driesvints committed:
Remove test skipping

driesvints committed:
Update tests.yml

driesvints committed:
Update tests.yml

driesvints committed:
Bump SwiftMailer

driesvints committed:
string fallbacks

driesvints committed:
Update phpunit.xml.dist

driesvints committed:
Remove skips
